DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/CourseSetup/CourseSetupControllerTests.cs

```diff
@@ -151,22 +219,446 @@ public void Index_with_CLEAR_filterBy_and_new_filter_query_parameter_sets_new_co
 
         [Test]
         public void Index_with_no_filtering_should_default_to_Active_courses()
+        {
+            // When
+            var result = controller.Index();
+
+            // Then
+            result.As<ViewResult>().Model.As<CourseSetupViewModel>().FilterBy.Should()
+                .Be("Status|Active|true");
+        }
+
+        [Test]
+        public void AddCourseNew_sets_new_temp_data()
+        {
+            // When
+            var result = controller.AddCourseNew();
+
+            // Then
+            controller.TempData.Peek<AddNewCentreCourseData>().Should().NotBeNull();
+            result.Should().BeRedirectToActionResult().WithActionName("SelectCourse");
+        }
+
+        [Test]
+        public void SelectCourse_post_updates_temp_data_and_redirects()
+        {
+            var applicationOptionsSelectList = new List<SelectListItem> { new SelectListItem("Test Name", "1") };
+            var model = new SelectCourseViewModel(application.ApplicationId, applicationOptionsSelectList);
+            SetAddNewCentreCourseTempData();
+
+            // When
+            var result = controller.SelectCourse(model);
+
+            // Then
+            controller.TempData.Peek<AddNewCentreCourseData>()!.Application.Should()
+                .BeEquivalentTo(application);
+            controller.TempData.Peek<AddNewCentreCourseData>()!.SetCourseDetailsModel.Should().BeNull();
+            controller.TempData.Peek<AddNewCentreCourseData>()!.SetCourseOptionsModel.Should().BeNull();
+            controller.TempData.Peek<AddNewCentreCourseData>()!.SetCourseContentModel.Should().BeNull();
+            controller.TempData.Peek<AddNewCentreCourseData>()!.SetSectionContentModels.Should().BeNull();
+            result.Should().BeRedirectToActionResult().WithActionName("SetCourseDetails");
+        }
+
+        [Test]
+        public void SelectCourse_does_not_redirect_with_invalid_model()
+        {
+            var model = new SelectCourseViewModel { ApplicationId = 1 };
+            controller.ModelState.AddModelError("ApplicationId", "Select a course");
+            SetAddNewCentreCourseTempData(application);
+
+            // When
+            var result = controller.SelectCourse(model);
+
+            // Then
+            result.Should().BeViewResult().ModelAs<SelectCourseViewModel>();
+            controller.ModelState["ApplicationId"].Errors[0].ErrorMessage.Should()
+                .BeEquivalentTo("Select a course");

```

Comment:
Can use `Be(..)` here instead of `.BeEquivalentTo(..)`

Reply:
done